packages/components/MIGRATING.md

```diff
@@ -66,6 +69,105 @@ Changes:
+
+    -   _required_ - The `Dialog.Trigger` is used to pass in a component to trigger the dialog.
+
+    -   _required_ - The `Dialog.Content` is the container appearing when the dialog is Visible
+
+        -   _optional_ - The `Dialog.Header` can be passed in as a child of `Dialog.Content` to add a styled header to the dialog.
+
+        -   _optional_ - The `Dialog.Body` can be passed in as a child of `Dialog.Content` to add a styled body to the dialog.
+
+        -   _optional_ - The `Dialog.Footer` can be passed in as a child of `Dialog.Content` to add a styled footer to the dialog.
+
+        -   _optional_ - The `Dialog.SideContent` can be passed in as a child of `Dialog.Content` to add a space on the left side of the dialog. Custom components can then be passed in as as children of `Dialog.SideContent`
+
+-   The styling / layout is now controlled on the subcomponent `Dialog.Content`
+
+-   The subcomponent `Dialog.Content` is used to display the dialog container.
+    It provides no styling by default and can be used to wrap a custom content.
+
+    -   `roundedCorners` is now called `rounded` and is a boolean.
+
+    -   The `width` and `maxHeight` props were removed. The Dialog container adjusts to the content inside. You can use a custom component inside if neccessary. you can use `minWidth`, `maxWidth` and `minHeight` to override the default values.
+
+    -   The `strategy` and `role` props were removed to simplify the API.
+
+    -   The `enablePortal` and `verticalAlignment` props were removed to standardize the dialog.
+
+        The dialog now uses a portal by default and is placed centered
+
+-   The subcomponents `Dialog.Header`, `Dialog.Body`, `Dialog.Footer` and `Dialog.SideContent` can be used to add defaut styling to the content inside of `Dialog.Content`.
+
+#### Old
+
+```tsx
+const [isOpen, setIsOpen] = useState(false);
+
+return (
+    <Button icon={<IconIcon/>} ref={triggerRef} onClick={() => setIsOpen(!isOpen)}>
+        Dialog Trigger
+    </Button>
+
+    <Dialog {...args} anchor={triggerRef} open={isOpen} handleClose={() => setIsOpen(false)}>
+        <DialogBody padding="spacious">
+            <p>
+                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad illum impedit iure numquam praesentium vel.
+                Distinctio perferendis, suscipit! Dolor doloremque et ex, modi nobis officiis perspiciatis quis tempora
+                temporibus voluptates?
+            </p>
+        </DialogBody>
+        <DialogFooter
+            actionButtons={[
+                {
+                    children: 'Close',
+                    onClick: () => {},
+                },
+            ]}
+        ></DialogFooter>
+    </Dialog>
+)
+```
+
+#### New
+
+```tsx
+<Dialog.Root>
+    <Dialog.Trigger>
+        <Button>Dialog Trigger</Button>
+    </Dialog.Trigger>
+    <Dialog.Content {...args}>
+        <Dialog.SideContent>
```

Comment:
You can add `className` to `SideContent`, then the extra div is gone

Reply:
this div is just my way of putting content in it.

I would not allow classname because then it can impact the layout